dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/events/SmapEntryFactory.java

```diff
@@ -91,34 +86,49 @@ private static HashMap<Long, String> getAnnotatedRegions() {
         String[] emptyStringArgs = {};
         Object[] dcmdArgs = {emptyStringArgs};
         String[] signature = {String[].class.getName()};
-
         String[] lines =
             ((String) mbs.invoke(objectName, "systemMap", dcmdArgs, signature)).split("\n");
         HashMap<Long, String> annotatedRegions = new HashMap<>();
+        boolean arrivedAtMappings = false;
 
         for (String line : lines) {
-          Matcher matcher = SYSTEM_MAP_ENTRY_PATTERN.matcher(line);
-          if (matcher.matches()) {
-            long startAddress;
-            if (matcher.group(1).equals("0x" + VSYSCALL_START_ADDRESS)) {
-              // See how smap entry parsing is done for vsyscall
-              startAddress = -0x1000 - 1;
-            } else {
-              startAddress = Long.decode(matcher.group(1));
+          if (!arrivedAtMappings) {
+            if (line.startsWith("size")) {
+              arrivedAtMappings = true;
+            }
+            continue;
+          } else {
+            if (line.startsWith("Total")) {
+              break;
             }
-            String description = matcher.group(6);
-            annotatedRegions.put(startAddress, description);
-            if (description.isEmpty()) {
-              annotatedRegions.put(startAddress, "UNDEFINED");
-            } else if (description.startsWith("STACK")) {
+          }
+
+          String[] segments = line.split("\\s+");
+          long startAddress;
+          if (segments[0].equals("0x" + VSYSCALL_START_ADDRESS)) {
+            startAddress = -0x1000 - 1;
+          } else {
+            startAddress = Long.decode(segments[0]);
+          }
+
+          if (segments.length >= 7) {
+            String description = segments[6];
+            if (description.startsWith("STACK")) {
               annotatedRegions.put(startAddress, "STACK");
             } else if (description.startsWith("[") || description.startsWith("/")) {
               annotatedRegions.put(startAddress, "SYSTEM");
+            } else if (description.startsWith("CDS")) {
+              annotatedRegions.put(startAddress, "CDS");
+            } else if (description.startsWith("INTERN")) {
+              annotatedRegions.put(startAddress, "INTERN");
             } else {
-              annotatedRegions.put(startAddress, description.split("\\s+")[0]);
+              annotatedRegions.put(startAddress, description);
```

Comment:
Nit - it seems like all descriptors are upper case
```suggestion
              annotatedRegions.put(startAddress, description.toUpperCase());
```